cli/smoke.go

```diff
@@ -0,0 +1,204 @@
+	Conf   string `long:"conf" default:"" description:"Evergreen configuration file"`
+}
+
+// SmokeTestEndpointCommand runs tests against UI and API endpoints.
+type SmokeTestEndpointCommand struct {
+	TestFile string `long:"test-file" short:"t" description:"file with test endpoints definitions"`
+}
+
+// EndpointTestDefinitions describes the UI and API endpoints to verify are up.
+type EndpointTestDefinitions struct {
+	UI  map[string][]string `yaml:"ui,omitempty"`
+	API map[string][]string `yaml:"api,omitempty"`
+}
+
+// Execute starts the Evergreen web service and runner.
+func (c *StartEvergreenCommand) Execute(_ []string) error {
+	setSenderNameToSmoke()
+	wd, err := os.Getwd()
+	if err != nil {
+		return errors.Wrap(err, "error getting current directory")
+	}
+	if c.Binary == "" {
+		c.Binary = filepath.Join(wd, "clients", runtime.GOOS+"_"+runtime.GOARCH, defaultBinaryName)
+	}
+	if c.Conf == "" {
+		c.Conf = filepath.Join(wd, "scripts", defaultConfigFile)
+	}
+	web := exec.Command(c.Binary, "service", "web", "--conf", c.Conf)
+	web.Env = []string{fmt.Sprintf("EVGHOME=%s", wd), "PATH=" + strings.Replace(os.Getenv("PATH"), `\`, `\\`, -1)}
+	webSender := send.NewWriterSender(send.MakeNative())
+	defer webSender.Close()
+	webSender.SetName("web.service")
+	web.Stdout = webSender
+	web.Stderr = webSender
+
+	runner := exec.Command(c.Binary, "service", "runner", "--conf", c.Conf)
+	runner.Env = []string{fmt.Sprintf("EVGHOME=%s", wd), "PATH=" + strings.Replace(os.Getenv("PATH"), `\`, `\\`, -1)}
+	runnerSender := send.NewWriterSender(send.MakeNative())
+	defer runnerSender.Close()
+	runnerSender.SetName("runner")
+	runner.Stdout = runnerSender
+	runner.Stderr = runnerSender
+
+	if err := web.Start(); err != nil {
+		return errors.Wrap(err, "error starting web service")
+	}
+	if err := runner.Start(); err != nil {
+		return errors.Wrap(err, "error starting runner")
+	}
+
+	exit := make(chan error)
+	interrupt := make(chan os.Signal)
+	signal.Notify(interrupt, os.Interrupt)
+	go func() {
+		exit <- web.Wait()
+		grip.Errorf("web service exited: %s", err)
+	}()
+	go func() {
+		exit <- runner.Wait()
+		grip.Errorf("runner exited: %s", err)
+	}()
+
+	select {
+	case <-exit:
+		err = errors.New("problem running Evergreen")
+		grip.Error(err)
+		return err
+	case <-interrupt:
+		grip.Info("received SIGINT, killing Evergreen")
+		if err := web.Process.Kill(); err != nil {
+			grip.Errorf("error killing evergreen web service: %s", err)
+		}
+		if err := runner.Process.Kill(); err != nil {
+			grip.Errorf("error killing evergreen runner: %s", err)
+		}
+	}
+
+	return nil
+}
+
+// Execute runs tests against UI and API endpoints.
+func (c *SmokeTestEndpointCommand) Execute(_ []string) error {
+	setSenderNameToSmoke()
+	wd, err := os.Getwd()
+	if err != nil {
+		return errors.Wrap(err, "error getting current directory")
+	}
+
+	if c.TestFile == "" {
+		c.TestFile = filepath.Join(wd, "scripts", defaultTestFile)
```

Comment:
should we log to let users know that we're using the default?

Reply:
done